Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -777,11 +777,16 @@ protected function renderDraftContent(ilTemplate $tpl, string $action, bool $ren
 
         $frm = $this->object->Forum;
 
-        $draftsObjects = ilForumPostDraft::getInstancesByUserIdAndThreadId(
-            $this->user->getId(),
-            $this->objCurrentTopic->getId()
-        );
-        $drafts = $draftsObjects[$node->getId()];
+        if($_SESSION['viewmode'] >= ilForumProperties::VIEW_DATE) {
```

Comment:
Why do you use the `<` operator here? I'd expect something like `===` or `!=='.

Can't we use `$currentViewMode` instead of `$_SESSION['viewmode']`?

Reply:
No. Because the "$currentSortation === ilForumProperties::VIEW_TREE" condition does not match in every case when it is expected. 
I am still not sure why this happens, but the $currentSortation variable isn't even set properly in every changed viewmode case. 

The tests showed, that the session var is set properly instead.